needles/gnome_software_main_screen.json

```diff
@@ -20,9 +20,9 @@
     },
     {
       "": "Top bar buttons (updates button appears and disappears)",
-      "xpos": 321,
+      "xpos": 281,
```

Comment:
I don't think this will solve the problem. On https://openqa.dev.endlessm.com/tests/5867#step/install_flatpak/3 the exclusion area already fully covers all the buttons on both the needle and the actual image. I think the problem is that the date on the taskbar (which is not excluded) is wider, pushing the tray icons slightly to the left.

Reply:
(Is there some way we can globally exclude that bit of the taskbar?)

Reply:
tl;dr: This PR is definitely an improvement for the gnome-software tests, but doesn’t get them passing entirely.

I think it’s using the JSON from this branch when visualising the exclusion area on that page, which is an OpenQA bug (but I can’t see them doing any better — they’d have to archive a copy of the needles/JSON for each test run, indefinitely). The exclusion area was previously narrower.

The date has never actually been a problem. I think the default match percentage is 95%, and the date never changes more than 5% of pixels in the screenshot. The remaining problem on that test is the ‘USB’ category on the left, which I filed an issue for earlier: https://phabricator.endlessm.com/T24527

Reply:
To the best of my knowledge, there is no way to globally exclude the clock from screenshots. We’d have to copy the JSON snippet for excluding it into every JSON file. That would not be the end of the world.